sale_blanket_order/tests/test_sale_order.py

```diff
@@ -0,0 +1,198 @@
+                        0,
+                        {
+                            "product_id": self.product.id,
+                            "product_uom": self.product.uom_id.id,
+                            "date_schedule": fields.Date.to_string(
+                                self.date_schedule_2
+                            ),
+                            "original_uom_qty": 20.0,
+                            "price_unit": 30.0,
+                        },
+                    ),
+                ],
+            }
+        )
+        blanket_order.sudo().onchange_partner_id()
+        return blanket_order
+
+    def create_blanket_order_02(self):
+        blanket_order = self.blanket_order_obj.create(
+            {
+                "partner_id": self.partner.id,
+                "validity_date": fields.Date.to_string(self.validity),
+                "payment_term_id": self.payment_term.id,
+                "pricelist_id": self.sale_pricelist.id,
+                "line_ids": [
+                    (
+                        0,
+                        0,
+                        {
+                            "product_id": self.product.id,
+                            "product_uom": self.product.uom_id.id,
+                            "original_uom_qty": 20.0,
+                            "price_unit": 30.0,
+                        },
+                    ),
+                    (
+                        0,
+                        0,
+                        {
+                            "product_id": self.product_2.id,
+                            "product_uom": self.product.uom_id.id,
+                            "original_uom_qty": 20.0,
+                            "price_unit": 30.0,
+                        },
+                    ),
+                ],
+            }
+        )
+        blanket_order.sudo().onchange_partner_id()
+        return blanket_order
+
+    def test_01_create_sale_order(self):
+        blanket_order = self.create_blanket_order_01()
+        blanket_order.sudo().action_confirm()
+        bo_lines = self.blanket_order_line_obj.search(
+            [("order_id", "=", blanket_order.id)]
+        )
+        self.assertEqual(len(bo_lines), 2)
+
+        so = self.sale_order_obj.create(
+            {
+                "partner_id": self.partner.id,
+                "order_line": [
+                    (
+                        0,
+                        0,
+                        {
+                            "name": self.product.name,
+                            "product_id": self.product.id,
+                            "product_uom_qty": 5.0,
+                            "product_uom": self.product.uom_po_id.id,
+                            "price_unit": 10.0,
+                        },
+                    )
+                ],
+            }
+        )
+        so_line = so.order_line[0]
+        so_line.with_context(from_sale_order=True).name_get()
+        so_line.onchange_product_id()
```

Comment:
if you have time this test can be improved with class Form, it will play onchanges same as UI

Reply:
Hello I'm creating a new PR #1466 where I'm applyng all the suggestions. What to dou mean using class Form in unit test?

Reply:
Calling on_change methods in your tests is totally wrong, instead create the sale order using the Form (from odoo.tests import Form)
There are a lot of examples already in the Odoo / OCA source Code